src/services/oauth2/hooks/index.js

```diff
@@ -20,8 +20,15 @@ const setSubject = (hook) => {
 				isLocal: true,
 			},
 		})
-		.then((tools) =>
-			hook.app
+		.then((tools) => {
+			if (!tools.data || !Array.isArray(tools.data) || tools.data.length !== 1) {
+				throw new Error(
```

Comment:
In featherJS it exists simliar errors like we have from nest. BadRequest and so on. Maybe one of them is better then a generic js error.